include/fiction/algorithms/network_transformation/inverter_balancing.hpp

```diff
@@ -0,0 +1,320 @@
+                {
+                    old2new[g] = ntk_dest.create_node(children, ntk.node_function(g));
+                    return true;
+                }
+
+                return true;
+            });
+
+        ntk.foreach_po(
+            [this, &ntk_dest, &old2new](const auto& po)
+            {
+                auto tgt_signal = old2new[ntk.get_node(po)];
+                auto tgt_po     = ntk.is_complemented(po) ? ntk_dest.create_not(tgt_signal) : tgt_signal;
+                if (tgt_po == 0)
+                {
+                    tgt_signal = old2new[ntk.get_node(preserved_po[0])];
+                    preserved_po.erase(preserved_po.begin());
+                    tgt_po = ntk.is_complemented(po) ? ntk_dest.create_not(tgt_signal) : tgt_signal;
+                }
+                ntk_dest.create_po(tgt_po);
+            });
+
+        if constexpr (mockturtle::has_foreach_ri_v<TopoNtkSrc>)
+        {
+            ntk.foreach_ri(
+                [this, &ntk_dest, &old2new](const auto& po)
+                {
+                    const auto tgt_signal = old2new[ntk.get_node(po)];
+                    const auto tgt_po     = ntk.is_complemented(po) ? ntk_dest.create_not(tgt_signal) : tgt_signal;
+
+                    ntk_dest.create_ri(tgt_po);
+                });
+        }
+
+        // restore signal names if applicable
+        fiction::restore_names(ntk, ntk_dest, old2new);
+
+        return ntk_dest;
+    }
+
+  private:
+    using TopoNtkSrc = mockturtle::topo_view<Ntk>;
+    TopoNtkSrc                          ntk;
+    mockturtle::fanout_view<TopoNtkSrc> fo_ntk{ntk};
+    std::vector<mockturtle::node<Ntk>>  x_inv{};         // inverter nodes, which get deleted
+    std::vector<mockturtle::node<Ntk>>  m_inv{};         // inverter nodes, which get moved to fanin position
+    std::vector<mockturtle::node<Ntk>>  blc_fos{};       // fo nodes, where balancing is applied
+    std::vector<mockturtle::node<Ntk>>  preserved_po{};  // nodes where pos need to be preserved
+};
+
+template <typename NtkSrc>
+bool inverter_balancing_recursive(const NtkSrc& ntk)
+{
+    bool       return_val = false;
+    const auto fo_ntk     = mockturtle::fanout_view<NtkSrc>(ntk);
+    ntk.foreach_gate(
+        [&](const auto& g)
+        {
+            if (ntk.fanout_size(g) >= 2)
+            {
+                auto fanout_inv = fanouts(fo_ntk, g);
+                auto balance    = std::all_of(fanout_inv.begin(), fanout_inv.end(),
+                                              [&](const auto& fo_node) { return fo_ntk.is_inv(fo_node); });
+                if (balance && fanout_inv.size() > 1)
+                {
+                    return_val = true;
+                }
+            }
+        });
+    return return_val;
+}
+
+}  // namespace detail
+
+/**
+ * Balances inverters at fanout nodes. Thereby, this function is very similar to fiction::network_conversion.
```

Comment:
The conclusion here is not obvious to me as a reader. Why is inverter balancing similar to network conversion?